ingredient/ingredient-event-hub-namespace/src/functions.ts

```diff
@@ -2,11 +2,42 @@ import {BaseUtility, IngredientManager} from '@azbake/core'
 
 export class EventHubNamespaceUtils extends BaseUtility {
 
-    public create_resource_name(): string {
+    public get_resource_name(shortName: string | null = null): string {
         let util = IngredientManager.getIngredientFunction("coreutils", this.context)
 
-        const profile = util.create_resource_name("ehn", null, true);
-        return profile;
+        //resource type, name, region enabled
+        const fullName = util.create_resource_name("ehn", shortName, true);
+
+        this.context._logger.debug(`EventHubNamespaceUtils.get_resource_name() returned ${fullName}`);
+
+        return fullName;
     } 
+
+    public async get_resource_group(rgShortName: string | null = null): Promise<string> {        
+        let util = IngredientManager.getIngredientFunction("coreutils", this.context)
+
+        var rg: string;
+
+        let override = this.context.Config.rgOverride
+        if (override)
+            rg = await override.valueAsync(this.context);
+        //resource type, name, region enabled
+        else rg = util.create_resource_name("", rgShortName, true);
+
+        this.context._logger.debug(`EventHubNamespaceUtils.get_resource_group() returned ${rg}`);
+
+        return rg;
+    }
+    
+    public async get_resource_profile(shortName: string | null = null, rgShortName: string | null = null): Promise<string> {
```

Comment:
this should get moved to coreutils if we don't already have this over there. This isn't specific to event hubs, and I've seen this rg/resource combined string used in a few places now.

Reply:
I'd like to move to coreutils but it relies on the ingredient's local get_resource_name function that specifies the "ehn" resource type.  I can get it to work but coreutils would need info from the ingredient being deployed.  Thoughts?